### PR TITLE
Create docs navigation sidebar

### PR DESCRIPTION
**Pull request by corwintines**
- Adds left sidebar to navigate docs
- Adds right sidebar to navigate current document
- Adds Note.tsx component for notes in markdown

### CODE REVIEW

File: src/components/UI/docs/DocsLinks.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+      })}
+    </Stack>
+  );
+};
+
+interface LinksListProps {
+  links: any[];
+}
+
+const LinksList: FC<LinksListProps> = ({ links }) => {
```
Comment (nhsz) [inline src/components/UI/docs/DocsLinks.tsx:74]: should be moved to a separate UI component

File: src/data/documentation-links.yaml [hunk 2]
```diff
@@ -0,0 +1,149 @@
+- id: Getting started
```
Comment (nhsz) [inline src/data/documentation-links.yaml:1]: @corwintines this list is hardcoded, if we add/remove a doc or edit a path we must remember to keep this list updated which is not ideal.

**Review by nhsz — CHANGES_REQUESTED** [review]
@corwintines Looking good, left some comments. Right margin of the navbar [is bigger on the design](https://www.figma.com/file/ekzIgwyeVKLtFSAcnA0Q0D/geth-website?node-id=694%3A2580&t=ne2XHx6pIhbRMqgV-0)

**Review by wackerow — CHANGES_REQUESTED** [review]
Awesome job! Great start!

Noticing a layout bug with the border on the chevron toggle, which seems to be worse when text spans more than one line
image
image

Also when spot checking some pages, getting a 404 on this one:
image
...located [here](https://deploy-preview-40--melodious-puffpuff-8e1109.netlify.app/docs/developers/evm-tracing/build-in-tracers)
Also [here](https://deploy-preview-40--melodious-puffpuff-8e1109.netlify.app/docs/developers/evm-tracing/custom-tracers)

---

Left some suggestions re: removing `any` types in lieu of an appropriate interface, and some syntactic sugar suggestions like destructuring items when mapping and removing curly brackets/"return" for arrow functions that can use implicit returns. 

Overall looking great tough! 🤩

File: src/data/documentation-links.yaml (hunk 2 above)
Comment (corwintines) [reply src/data/documentation-links.yaml:1]: Porting over discord discussion here, the reason I opted for this was build the structure for the docs navbar

**Comment by corwintines** [comment]
> @corwintines Looking good, left some comments. Right margin of the navbar [is bigger on the design](https://www.figma.com/file/ekzIgwyeVKLtFSAcnA0Q0D/geth-website?node-id=694%3A2580&t=ne2XHx6pIhbRMqgV-0)

Not sure which margin you mean here

File: src/data/documentation-links.yaml (hunk 2 above)
Comment (wackerow) [reply src/data/documentation-links.yaml:1]: This is a similar approach to what is used on ethereum.org

**Review by wackerow — COMMENTED** [review]
So close! Still seeing a small layout shift by a couple pixels with these accordions
image

**Comment by nhsz** [comment]
> 

I see a different right margin on the design bu yeah can be related to resolution. Top margin of the content should be higher though, currentle the breadcrumbs appear above the level of the sidebar (figma image below)

Screen Shot 2022-11-30 at 10 44 13

**Comment by nhsz** [comment]
I think the breadcrumb (`docs` at the top left) should not appear when we are on the root `/docs` 

Screen Shot 2022-11-30 at 10 42 38

**Review by wackerow — APPROVED** [review]
Nice! The layout bug looks fixed, and the broken links are now working. "Notes" component looks good

Couple things I'm noting now:

If the markdown file starts with an h2, it's resulting in a lot of space between the h1 (from `title`) and the initial h2:
image

Also, should we make some indication of the active page on the left? I would think the name of the active page would be bold or a different color, but right now it's unstyled:
image

Don't think either of these necessarily need to be blockers though. If we don't fix them here we can open issues and address separately.

**Review by nhsz — CHANGES_REQUESTED** [review]
@corwintines Great! 🔥🔥🔥 Asked for a change on the file structure. Could you please add the instructions on `README` with the steps to add a new doc and updating `documentation-links`? Thx

**Review by nhsz — APPROVED** [review]
LGTM!

**Pull request merged**